takwimu/management/commands/update_topics_index.py

```diff
@@ -25,45 +34,127 @@ def index_topics(self, queryset, search_backend):
             for topic in i.body.stream_data:
                 topic_id = topic['id']
                 title = topic['value'].get('title', '')
-                topic_body = topic['value'].get('body', '')
                 topic_summary = topic['value'].get('summary', '')
-                body = '\n'.join([topic_summary, topic_body])
-                metadata = ''
-
-                _, outcome = search_backend.add_to_index(topic_id,
-                                                         'topic',
-                                                         country,
-                                                         category,
-                                                         title,
-                                                         body,
-                                                         metadata,
-                                                         parent_page_id,
-                                                         parent_page_type,
-                                                         )
-                self.stdout.write(
-                    search_backend.es_index + ": Indexing topic '%s result %s'" % (
-                        topic_id,
-                        outcome,
-                    ))
-
-                indicators = topic['value'].get('indicators', '')
-                for indicator in indicators:
-                    for widget in indicator['value']['widgets']:
-                        data = get_widget_data(widget)
-                        if data:
-                            _, outcome = search_backend.add_to_index(
-                                data['id'],
-                                'indicator_widget',
-                                country,
-                                category,
-                                data['title'],
-                                data['body'],
-                                data['metadata'],
-                                parent_page_id,
-                                parent_page_type
-                            )
-                            self.stdout.write(
-                                search_backend.es_index + ": Indexing widget '%s result %s'" % (
+                topic_body = topic['value'].get('body', '')
+                for k in topic_body:
+                    if k['type'] == 'text':
```

Comment:
```
Traceback (most recent call last):
  File "manage.py", line 22, in <module>
    execute_from_command_line(sys.argv)
  File "/usr/local/lib/python3.7/site-packages/django/core/management/__init__.py", line 364, in execute_from_command_line
    utility.execute()
  File "/usr/local/lib/python3.7/site-packages/django/core/management/__init__.py", line 356, in execute
    self.fetch_command(subcommand).run_from_argv(self.argv)
  File "/usr/local/lib/python3.7/site-packages/django/core/management/base.py", line 283, in run_from_argv
    self.execute(*args, **cmd_options)
  File "/usr/local/lib/python3.7/site-packages/django/core/management/base.py", line 330, in execute
    output = self.handle(*args, **options)
  File "/src/takwimu/takwimu/management/commands/update_topics_index.py", line 25, in handle
    self.index_topics(ProfileSectionPage.objects.live(), topic_search)
  File "/src/takwimu/takwimu/management/commands/update_topics_index.py", line 40, in index_topics
    if k['type'] == 'text':
TypeError: string indices must be integers
Makefile:17: recipe for target 'update_index' failed
make: *** [update_index] Error 1
```

@phillipahereza

Reply:
Could be the **old** topic block though... Let me rebuild and try again.

Reply:
@kilemensi probably this error `TypeError: string indices must be integers` is from that old topic block

Reply:
Works @phillipahereza